### PR TITLE
Tanisha assets

### PR DESCRIPTION
**Pull request by tanishagarg2**
Modified pDmgDealer and Peashooter class

### CODE REVIEW

File: src/assets/Plant.java [hunk 1]
```diff
@@ -1,32 +1,59 @@
...
-	public void takeDamage(int dmg) {
-		this.hp = hp - dmg; 
+	public void takeDamage(int dmg)	{
+		this.hitPoints -= dmg;
+		isAlive();
 	}
 	
-
+	public boolean isAlive() {
+		if(getHP() == 0 || getHP() < 0) {
```
Comment (dershao) [inline src/assets/Plant.java]: `getHP() <= 0`

File: src/assets/Zombie.java [hunk 2]
```diff
@@ -1,30 +1,76 @@
...
 	public void takeDamage(int dmg) {
-		// TODO Auto-generated method stub
-		
+		this.hitPoints -= dmg;
+		isAlive();
 	}
 
-
+	public boolean isAlive() {
+		if(getHP() == 0 || getHP() < 0) {
```
Comment (dershao) [inline src/assets/Zombie.java]: `getHP() <= 0`

File: src/assets/Plant.java [hunk 3]
```diff
@@ -1,32 +1,59 @@
 package assets;
 
+import util.Logger;
+
 public class Plant implements Unit{
-	public int speed = 0;
-	public int hp = 2;
-	public int dmg = 0;
+	private static Logger LOG = new Logger("Zombie");
+	private int speed;
```
Comment (DwangDistracted) [inline src/assets/Plant.java]: wouldn't this always be zero? why does this need to be an attribute

File: src/assets/Zombie.java [hunk 4]
```diff
@@ -1,30 +1,76 @@
...
 	@Override
 	public void takeDamage(int dmg) {
-		// TODO Auto-generated method stub
-		
+		this.hitPoints -= dmg;
+		isAlive();
 	}
 
-
+	public boolean isAlive() {
```
Comment (dershao) [inline src/assets/Zombie.java]: Could this be shared with both Plants and Zombies? (i.e., put in Unit class)

File: src/assets/Plant.java [hunk 5]
```diff
@@ -1,32 +1,59 @@
 package assets;
 
+import util.Logger;
+
 public class Plant implements Unit{
-	public int speed = 0;
-	public int hp = 2;
-	public int dmg = 0;
+	private static Logger LOG = new Logger("Zombie");
```
Comment (DwangDistracted) [inline src/assets/Plant.java]: the logger should be for a plant not zombie

File: src/assets/Plant.java [hunk 6]
```diff
@@ -1,32 +1,59 @@
 package assets;
 
+import util.Logger;
+
 public class Plant implements Unit{
```
Comment (dershao) [inline src/assets/Plant.java]: For consistency, should Plant also know it's own coordinates as well?

File: src/assets/pDmgDealer.java [hunk 7]
```diff
@@ -1,7 +1,10 @@
 package assets;
 /*
- * used to group plants that deal dmg
+ * used to group plants that deal have power to kill zombies
  */
 public class pDmgDealer extends Plant {
```
Comment (DwangDistracted) [inline src/assets/pDmgDealer.java]: I don't think we need this unless Dmg Dealers have different attributes from plants. maybe dmg should be moved from plant to this class?

File: src/assets/Flower.java [hunk 8]
```diff
@@ -1,31 +1,11 @@
 package assets;
 
 public class Flower extends Plant{
```
Comment (dershao) [inline src/assets/Flower.java]: Should we have Flower to know the amount of resources it can give per turn?

**Pull request merged**